src/main/java/com/redhat/devtools/lsp4ij/features/selectionRange/LSPExtendWordSelectionHandler.java

```diff
@@ -71,89 +57,36 @@ public boolean canSelect(@NotNull PsiElement element) {
     }
 
     @Override
-    public @Nullable List<TextRange> select(@NotNull PsiElement element,
-                                            @NotNull CharSequence editorText,
-                                            int offset,
-                                            @NotNull Editor editor) {
+    @Nullable
+    public List<TextRange> select(@NotNull PsiElement element,
+                                  @NotNull CharSequence editorText,
+                                  int offset,
+                                  @NotNull Editor editor) {
         PsiFile file = element.getContainingFile();
         if (file == null) {
             return null;
         }
 
         // Get the selection ranges
         Document document = editor.getDocument();
-        List<SelectionRange> selectionRanges = getSelectionRanges(file, document, offset);
+        List<SelectionRange> selectionRanges = LSPSelectionRangeSupport.getSelectionRanges(file, document, offset);
         if (ContainerUtil.isEmpty(selectionRanges)) {
             return null;
         }
 
         // Convert the selection ranges into text ranges
         Set<TextRange> textRanges = new LinkedHashSet<>(selectionRanges.size());
         for (SelectionRange selectionRange : selectionRanges) {
-            TextRange selectionTextRange = getTextRange(selectionRange, document);
+            TextRange selectionTextRange = LSPSelectionRangeSupport.getTextRange(selectionRange, document);
             textRanges.addAll(expandToWholeLinesWithBlanks(editorText, selectionTextRange));
 
             for (SelectionRange parentSelectionRange = selectionRange.getParent();
                  parentSelectionRange != null;
                  parentSelectionRange = parentSelectionRange.getParent()) {
-                TextRange parentSelectionTextRange = getTextRange(parentSelectionRange, document);
+                TextRange parentSelectionTextRange = LSPSelectionRangeSupport.getTextRange(parentSelectionRange, document);
                 textRanges.addAll(expandToWholeLinesWithBlanks(editorText, parentSelectionTextRange));
             }
         }
         return new ArrayList<>(textRanges);
     }
-
```

Comment:
Moved into `LSPSelectionRangeSupport` so that it can be used by `LSPCodeBlockProvider`.